### DISCUSSION TITLE
["⛔ Twitter Login attempt failed: Unknown subtask ArkoseLogin"]

### DISCUSSION THREAD

**Discussion by David-IChunLin** [elizaOS/eliza#1240] [Q&A]
Issue:
I'm encountering an issue while attempting to log into Twitter. The error message "Unknown subtask ArkoseLogin" suggests a problem with the authentication process:

["◎ Twitter client started"] 

 ["◎ Waiting for Twitter login"] 

 ["⛔ Login attempt failed: Unknown subtask ArkoseLogin"] 

 ["⛔ Failed to login to Twitter. Retrying... (4 attempts left)"] 

wondering: 
1. What could be causing the "Unknown subtask ArkoseLogin" error?
2. Are there any known workarounds or solutions for this specific error?
3. VPN related? 
 
Asking for Help:
If anyone has encountered a similar issue, please share your knowledge and experience.

**Comment by Caijiawen** [comment]
I get this error too

**Comment by alphastack1** [comment]
I found that the X API access works much better than just the email id pw. I saw they also recently removed the twitter_cookies for some reason..

**Reply by vedant-asati** [reply]
but i could not see x api field in env

**Comment by wildanzrrr** [comment]
I encountered this error when I was connected to a VPN. In my case I was using WARP, and the error appeared during the connection. However, once I disconnected from the VPN, everything returned to normal.

**Comment by hugoroussel** [comment]
Hi I'm getting this error as well

**Comment by vedant-asati** [comment]
I'd also faced the error, then I marked my x account as automated in the settings and it got fixed. It might happen due to x flagging your login attempt as spam.

**Reply by Stephen-Kimoi** [reply]
I also worked on my end when I made this change. But I am still getting the same error for the [twitter scrapper finetune](https://github.com/elizaOS/twitter-scraper-finetune).

**Comment by Lama1403** [comment]
I stiill get this error. It worked for a while then didn't anymore, any ideas?

**Comment by nvsd** [comment]
I get this error when tethering to my phone. Connecting to a VPN causes it to work. I'd just switch around connections in your VPN if you have one.

**Comment by divine-comedian** [comment]
I'm also facing this issue - I've tried enabling and disabling 2fa - marked my account as automated, tried various vpn connections and without vpn 

Sometimes I get 

```
 ["⛔ Login attempt failed: {\"errors\":[{\"code\":399,\"message\":\"Incorrect. Please try again. g;173828646685907473:-1738286466980:QAArMgO93X7smeZUUdqllRcb:6\"}]}"] 
```

or sometimes I get the same error OP mentioned.

**Reply by anciri** [reply]
Try this. I had the same problem, and it worked in my case

1/ go to the agent's twitter -> settings -> security and account access -> security -> two-factor authentication
2/ enable Authentication App
3/ click the button under the QR code to view it as string (it's 16 uppercase letters and numbers) - this is your 2FA SECRET
4/ complete the 2fa setup with an app like google authenticator
5/ add the 2fa secret to your .env file like this:
TWITTER_2FA_SECRET="YOUR_SECRET_CODE" (the code form point 3)

**Comment by divine-comedian** [comment]
I resolved this issue easily. I was actually logged in to the same twitter account on my desktop while trying to run the agent. 

I was able to proceed by logging myself out of twitter account on my desktop browser then running the startup script for the agent

**Reply by anciri** [reply]
It makes much more sense.  Cool! 
Anyway i have to try  if by doing what i did it allows me to stay logged in into the agent twitter account in my browser while running it.

**Comment by alohamonius** [comment]
Hey guys, got the same
Tried logout + MFA, same error(

**Comment by Prem95** [comment]
Same issue here

**Comment by cahrll** [comment]
same issue here on the eliza-starter, how to fix pls

**Comment by jongan69** [comment]
Use cookies, I have a few example repos that are working using cookie authentication

login to x and open developer console and memory

You need the 

auth_token, 
ct0,
twid

Check out this for some help:
https://github.com/jongan69/readEmailAndTweet/blob/main/tweet.js
